### PR TITLE
Fix set semantics bug with other store interface

### PR DESCRIPTION
**Pull request by maartyman**
@rubensworks is this what you had in mind?

### CODE REVIEW

File: lib/StreamingStore.ts [hunk 1]
```diff
@@ -5,6 +5,15 @@ import type { Readable } from 'readable-stream';
 import { PassThrough } from 'readable-stream';
 import { PendingStreamsIndex } from './PendingStreamsIndex';
 
+interface ILocalStore<Q extends RDF.BaseQuad> extends RDF.Store<Q> {
+  countQuads: (
+    subject: RDF.Term | null,
+    predicate: RDF.Term | null,
+    object: RDF.Term | null,
+    graph: RDF.Term | null,
```
Comment (rubensworks) [inline lib/StreamingStore.ts:13]:
```suggestion
    subject?: RDF.Term | null,
    predicate?: RDF.Term | null,
    object?: RDF.Term | null,
    graph?: RDF.Term | null,
```

**Review by rubensworks — CHANGES_REQUESTED** [review]
Yes, this looks perfect to me!

@surilindur could you try this out in link-traversal master and your void branch?

File: lib/StreamingStore.ts (hunk 1 above)
Comment (maartyman) [reply lib/StreamingStore.ts:13]: n3 store doesn't support `undefined`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4c017214b1c3fcfd5bb1637035f806b06c6b76d3/types/n3/index.d.ts#L295 the [OTerm](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4c017214b1c3fcfd5bb1637035f806b06c6b76d3/types/n3/index.d.ts#L128C13-L128C18) can only be a `RDF.Term`, string or null.

Comment (rubensworks) [reply lib/StreamingStore.ts:13]: It actually does support `undefined`: https://github.com/rdfjs/N3.js/blob/main/src/N3Store.js#L445-L447
But it appears the typings are not fully correct.

So yeah, let's keep it as-is for now. We can modify later once the N3 typings would get fixed.

Comment (maartyman) [reply lib/StreamingStore.ts:13]: I assume this is support for when [OTerm](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4c017214b1c3fcfd5bb1637035f806b06c6b76d3/types/n3/index.d.ts#L128C13-L128C18) is `null` but the logic is the same for `undefined`.

Comment (rubensworks) [reply lib/StreamingStore.ts:13]: Indeed, but since `null` is not the same as `undefined`, tsc will complain about it.

**Comment by surilindur** [comment]
This seems to produce the same results as the current 1.1.2 version in link traversal master branch, so it definitely does not seem to break that this time. I am also fairly sure I tested the correct versions of everything this time. Edit: also the tests in link traversal pass.

I will try to test with the VoID stuff soon, either today or tomorrow.

**Comment by surilindur** [comment]
After testing the version from this branch with the VoID-related work, it also seems to consistently produce the same results every time, with no spurious duplicates, since I cannot seem to be able to reproduce the cases where it was producing random numbers of results or exactly double the expected.

**Comment by rubensworks** [comment]
Excellent, thanks for checking @surilindur, and thanks for the fix @maartyman!

**Pull request merged**